crypto/rsa/rsa_backend.c

```diff
@@ -64,21 +64,34 @@ static int collect_numbers(STACK_OF(BIGNUM) *numbers,
 int ossl_rsa_fromdata(RSA *rsa, const OSSL_PARAM params[], int include_private)
 {
     const OSSL_PARAM *param_n, *param_e,  *param_d = NULL;
+    const OSSL_PARAM *param_derive = NULL;
     BIGNUM *n = NULL, *e = NULL, *d = NULL;
     STACK_OF(BIGNUM) *factors = NULL, *exps = NULL, *coeffs = NULL;
     int is_private = 0;
+    int derive_from_pq = 0;
+    BN_CTX *ctx = NULL;
 
     if (rsa == NULL)
         return 0;
 
+    ctx = BN_CTX_new_ex(rsa->libctx);
+    if (ctx == NULL)
+        return 0;
+
+    param_derive = OSSL_PARAM_locate_const(params, OSSL_PKEY_PARAM_RSA_DERIVE_FROM_PQ);
     param_n = OSSL_PARAM_locate_const(params, OSSL_PKEY_PARAM_RSA_N);
     param_e = OSSL_PARAM_locate_const(params, OSSL_PKEY_PARAM_RSA_E);
     if (include_private)
         param_d = OSSL_PARAM_locate_const(params, OSSL_PKEY_PARAM_RSA_D);
 
     if ((param_n != NULL && !OSSL_PARAM_get_BN(param_n, &n))
         || (param_e != NULL && !OSSL_PARAM_get_BN(param_e, &e))
-        || (param_d != NULL && !OSSL_PARAM_get_BN(param_d, &d)))
+        || (param_d != NULL && !OSSL_PARAM_get_BN(param_d, &d))) {
+            ERR_raise(ERR_LIB_RSA, ERR_R_PASSED_NULL_PARAMETER);
```

Comment:
I think the ERR_R_PASSED_NULL_PARAMETER is slightly misleading and I'd use ERR_R_PASSED_INVALID_ARGUMENT instead

Reply:
This does not seem to be resolved?

Reply:
IIRC I made the argument that n, d, and e were required parameters, not having them implied it was more accurate to indicate there was a NULL (implied missing) parameter, rather than a invalid (present but not correct) parameter

Reply:
Well... this failure is not necessarily about NULL parameter but about failure in OSSL_PARAM_get_BN() which can have various reasons including allocation failure. So IMO ERR_R_PASSED_NULL_PARAMETER is potentially misleading. One option would be to just remove this ERR_raise altogether as OSSL_PARAM_get_BN() will raise an error. But anyway.